core/src/main/java/com/sap/ai/sdk/core/common/RequestLogContext.java

```diff
@@ -0,0 +1,140 @@
+package com.sap.ai.sdk.core.common;
+
+import com.google.common.annotations.Beta;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.HttpEntity;
+import org.slf4j.MDC;
+
+/**
+ * Utility for managing MDC (Mapped Diagnostic Context) for logging of AI Core requests.
+ *
+ * <p>This class is intended for internal use only.
+ */
+@Slf4j
+@UtilityClass
+@Beta
+public class RequestLogContext {
```

Comment:
(Comment)

I'm okay with the API. You're marking it public and unstable - allowing us to use it across modules. Good!

Maybe I would prefer a proper singleton API like `RequestLogContext.getInstance().setXyz(...)` in combination with a mockable / replaceable logic for UUID generation.

These will be necessary if we ever decide to mock, test or allow-customizing the API.
However I wouldn't see this necessary for your initial PR here. No action required.

--- 

I was considering to suggest fluent API. But I feel maybe it's too soon for that, let's first find out whether current API usage fits our expectations and requirements.